Sources/Card/CardClient.swift

```diff
@@ -27,31 +25,31 @@ public class CardClient {
     ///   - orderID: The ID of the order to be approved
     ///   - card: The card to be charged for this order
     ///   - completion: Completion handler for approveOrder, which contains data of the order if success, or an error if failure
-    public func approveOrder(orderID: String, card: Card, completion: @escaping (Result<OrderData, Error>) -> Void) {
-        do {
-            let confirmPaymentRequest = try ConfirmPaymentSourceRequest(
-                card: card,
-                orderID: orderID,
-                clientID: config.clientID
-            )
+    public func approveOrder(
+        orderID: String,
+        card: Card,
+        completion: @escaping (Result<OrderData, NetworkingError>, CorrelationID?) -> Void
+    ) {
+        let path = "/v2/checkout/orders/\(orderID)/confirm-payment-source"
+        var request = APIRequest(method: .post, path: path)
+        request.body = ConfirmPaymentSourceBody(card: card)
 
-            apiClient.fetch(endpoint: confirmPaymentRequest) { result, _ in
-                switch result {
-                case .success(let response):
-                    let orderData = OrderData(orderID: response.id, status: OrderStatus(rawValue: response.status))
-                    completion(.success(orderData))
+        api.send(request) { response in
+            let correlationID = response.headers["Paypal-Debug-Id"] as? String
 
-                case .failure(let error):
-                    completion(.failure(error))
+            if response.status == 404 {
+                completion(.failure(NetworkingError.unknown), correlationID)
+            } else if let body = response.body {
+                do {
+                    let result = try self.decoder.decode(ConfirmPaymentSourceResponse.self, from: body)
```

Comment:
Lifting the responsibility of decoding the response object from `APIClient` to the feature client allows us to keep response objects within their respective modules without having to use generics.

Reply:
I can see why moving the parsing logic into each module makes sense/ is helpful. Especially for weirder scenarios like the EmptyResponse case, or once we had a specific 4XX code come back from a PayPal endpoint that actually had success/ result data we needed to parse to continue the flow (weird - but happened).

Reply:
is it very difficult to do that with current implementation? 🤔

Reply:
It seems like Alamo may have introduced this [out of necessity](https://github.com/Alamofire/Alamofire/issues/2466) for non-standard API responses. They originally inferred an empty response with the `204 No Content` HTTP status code. I'd expect the Orders API holds up a lot of these HTTP best practices.

Reply:
I see [they use generics](https://github.com/Alamofire/Alamofire/blob/effb907b0e27bbdff1d773c39346e1bac319fc11/Source/ResponseSerialization.swift#L961) for response types -> check if data exists -> if yes try decoding the generic type from data, if no (which by default is for 204 205 for them) return an instance of the `EmptyResponse`, sounds pretty much like the logic we have.

Do you see there is any specific downside from having/handling `EmptyResponse`?

Reply:
I see. I think it makes sense for Alamofire conceptually they have to support any restful API a developer wishes to connect to. In our case, we're only talking to orders at the moment. Having an additional type for this scenario is also adding additional complexity.

Reply:
The way that alamofire is using generics matches what we are using generics for though, doesn't it? in both cases, the goal is to provide a typed response to whoever is making a network request. 

we may have to interact with an API where we need to handle the url response in an unconventional way - and in that case i think we could make use of another fetch function (was trying to get at that [on the original networking PR here](https://github.com/paypal/iOS-SDK/pull/9/files#r713174796)). but i don't think we should go out of our way to support that until we know that we will need to